### PR TITLE
Add PlugX rule proposed in #469

### PR DESCRIPTION
**Pull request by Still34**
# Summary

This PR adds the first malware-specific capa rule as initially proposed in issue #469. Specifically, this PR adds the known module watermarks found in various builds of PlugX.

### CODE REVIEW

**Comment by Still34** [comment]
Per the linter, guess upstream needs to be updated to support the `maec` meta field.

File: malware/plugx/match-known-module.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+rule:
+  meta:
+    name: match known module
+    namespace: malware/plugx
```
Comment (williballenthin) [inline malware/plugx/match-known-module.yml:4]:
```suggestion
    namespace: malware-family/plugx
```

and change file path

**Review by williballenthin — APPROVED** [review]
looks great to me, thanks!

only one suggestion to tweak the directory path for family rules. let's use "malware-family" instead of "malware". with that change, i'll be happy to merge.

File: malware/plugx/match-known-module.yml (hunk 1 above)
Comment (Still34) [reply malware/plugx/match-known-module.yml]: Done 🙌

File: malware-family/plugx/match-known-module.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    name: match known module
+    namespace: malware-family/plugx
+    author: still@teamt5.org
+    scope: function
+    references:
+      - https://circl.lu/assets/files/tr-12/tr-12-circl-plugx-analysis-v1.pdf
+      - https://www.fireeye.com/blog/threat-research/2014/07/pacific-ring-of-fire-plugx-kaba.html
+      - https://www.avira.com/en/blog/new-wave-of-plugx-targets-hong-kong
+    examples:
+      - 64E9F62840DB2F65FC717CFAF99081F9:0x10024950
```
Comment (mr-tz) [inline malware-family/plugx/match-known-module.yml:12]: can you add this sample to capa-testfiles, please?

File: malware-family/plugx/match-known-module.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
+rule:
+  meta:
+    name: match known module
```
Comment (mr-tz) [inline malware-family/plugx/match-known-module.yml:3]:
```suggestion
    name: match known PlugX module
```

Comment (mr-tz) [reply malware-family/plugx/match-known-module.yml:3]: I think the name should include the malware family or at least be a little bit more descriptive.

File: malware-family/plugx/match-known-module.yml (hunk 2 above)
Comment (Still34) [reply malware-family/plugx/match-known-module.yml:12]: Created PR @ https://github.com/mandiant/capa-testfiles/pull/122

File: malware-family/plugx/match-known-module.yml (hunk 3 above)
Comment (Still34) [reply malware-family/plugx/match-known-module.yml]: Added a `description` meta field along with the suggested actions above.

File: malware-family/plugx/match-known-module.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+rule:
+  meta:
+    name: match known PlugX module
+    description: the sample references known PlugX watermarks (hexified YYYYMMDD + command opcode)
+    namespace: malware-family/plugx
+    author: still@teamt5.org
```
Comment (mr-tz) [inline malware-family/plugx/match-known-module.yml:6]:
```suggestion
    namespace: malware-family/plugx
    author: still@teamt5.org
    description: the sample references known PlugX watermarks (hexified YYYYMMDD + command opcode)
```

File: malware-family/plugx/match-known-module.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
+rule:
+  meta:
+    name: match known PlugX module
```
Comment (mr-tz) [inline malware-family/plugx/match-known-module.yml:3]: thanks, the file needs to be renamed accordingly

**Review by mr-tz — COMMENTED** [review]
almost good to go, sorry for the annoyance with the little fixes

**Comment by Still34** [comment]
No problem. Hope the changes above are good now!

File: malware-family/plugx/match-known-plugx-module.yml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        - number: 0x9007 = get registry value
+        - number: 0xA000 = enumerate network resources
+        - number: 0xB000 = start port mapping
+        - number: 0xC000 = get sql data source information
+        - number: 0xC001 = get sql driver description
+        - number: 0xC002 = execute sql statement
+        - number: 0xD000 = get TCP table
+        - number: 0xD001 = get UDP table
+        - number: 0xD002 = set TCP entry
+        - number: 0xE000 = start keylogger thread
```
Comment (mr-tz) [inline malware-family/plugx/match-known-plugx-module.yml:77]:
```suggestion
        - number: 0xE000 = start keylogger thread

```

File: malware-family/plugx/match-known-plugx-module.yml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    author: still@teamt5.org
+    description: the sample references known PlugX watermarks (hexified YYYYMMDD + command opcode)
+    scope: function
+    references:
+      - https://circl.lu/assets/files/tr-12/tr-12-circl-plugx-analysis-v1.pdf
+      - https://www.fireeye.com/blog/threat-research/2014/07/pacific-ring-of-fire-plugx-kaba.html
+      - https://www.avira.com/en/blog/new-wave-of-plugx-targets-hong-kong
+    examples:
+      - 64E9F62840DB2F65FC717CFAF99081F9:0x10024950
+    maec/malware-family: PlugX
```
Comment (mr-tz) [inline malware-family/plugx/match-known-plugx-module.yml:14]:
```suggestion
    maec/malware-family: PlugX
    author: still@teamt5.org
    description: the sample references known PlugX watermarks (hexified YYYYMMDD + command opcode)
    scope: function
    references:
      - https://circl.lu/assets/files/tr-12/tr-12-circl-plugx-analysis-v1.pdf
      - https://www.fireeye.com/blog/threat-research/2014/07/pacific-ring-of-fire-plugx-kaba.html
      - https://www.avira.com/en/blog/new-wave-of-plugx-targets-hong-kong
    examples:
      - 64E9F62840DB2F65FC717CFAF99081F9:0x10024950
```

**Review by mr-tz — CHANGES_REQUESTED** [review]
final formatting issues (hopefully)

**Comment by mr-tz** [comment]
awesome, thank you very much!

**Pull request merged**